lib/ansible/plugins/action/package.py

```diff
@@ -45,14 +47,22 @@ def run(self, tmp=None, task_vars=None):
         if module == 'auto':
             try:
                 if self._task.delegate_to:  # if we delegate, we should use delegated host's facts
```

Comment:
Assuming the fact run is working, doesn't doing the override logic first preclude a custom fact module from setting this? Maybe that's a feature, not a bug, but we should probably be explicit about that (and enforce it with a test?) if it's the desired behavior.

Reply:
i specified now in docs how it works, but our own 'fact finding' still uses setup.py directly  .. .not sure i want to use gather_facts as it is not guaranteed we'll get pgk_mgr info if they customized it, nor do i want to make it x2 expensive and fallback to setup .. but could ...